### PR TITLE
[Dynamo] Improve BuiltinVariable log when incorrect arg count happens

### LINKED ISSUE #1832: incorrect arg count <bound method BuiltinVariable.call_dict of BuiltinVariable(dict)> missing a required argument: 'arg'
### 🐛 Describe the bug

call dict throws warning incorrect arg count <bound method BuiltinVariable.call_dict of BuiltinVariable(dict)> missing a required argument: 'arg'

### Error logs

torch._dynamo.variables.builtin: [WARNING] incorrect arg count <bound method BuiltinVariable.call_dict of BuiltinVariable(dict)> missing a required argument: 'arg'
{}

### Minified repro

```
import torch._dynamo

@torch._dynamo.optimize('eager')
def test_call_dict():
    a = dict()
    return a

res = test_call_dict()
print(res)
```

### PR DESCRIPTION
**Pull request by yanboliang**
Fixes https://github.com/pytorch/torchdynamo/issues/1832

cc @mlazos @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

### CODE REVIEW

File: torch/_dynamo/variables/builtin.py [hunk 1]
```diff
@@ -326,7 +326,10 @@ def call_function(
             try:
                 inspect.signature(handler).bind(tx, *args, **kwargs)
             except TypeError as exc:
-                log.warning(f"incorrect arg count {handler} {exc}")
+                if not has_constant_handler:
```
Comment (SeaOfOcean) [inline torch/_dynamo/variables/builtin.py:329]: seems if `if has_constant_handler:` branch is executed before `if handler`, then this warning for dict call can be avoided.

Comment (yanboliang) [reply torch/_dynamo/variables/builtin.py:329]: Actually no, we get the value of ```has_constant_handler``` at [here](https://github.com/pytorch/pytorch/blob/master/torch/_dynamo/variables/builtin.py#L234), but executing the constant handler at [later](https://github.com/pytorch/pytorch/blob/master/torch/_dynamo/variables/builtin.py#L343).

**Review by mlazos — APPROVED** [review]
Is this error message that much more informative? What does it mean to not have a constant handler? 

LGTM though.

**Comment by yanboliang** [comment]
> Is this error message that much more informative? What does it mean to not have a constant handler?

I think so. Previously, it prints out warning even there is a constant handler(not fallback to python), which may confuse users like @SeaOfOcean. After this PR, if users see this warning, that means a fallback either because of ```incorrect arg count``` or ```no constant handler```.

**Pull request closed without merging**